playground/frontend/src/components/console.js

```diff
@@ -0,0 +1,76 @@
+import React, { 
+  useState, 
+  useEffect,
+} from 'react';
+import { 
+  Hook, Unhook, Console as ConsoleLib, Decode,
+} from 'console-feed';
+
+import { Grid, Paper, makeStyles } from '@material-ui/core';
+
+import ConsoleHeader from './consoleheader';
+
+const useStyles = makeStyles(theme => ({
+  wrapper: {},
+  header: {
+    padding: theme.spacing(2),
+  },
+}));
+
+/**
+ * Sets up a hook to capture all console method calls
+ * Don't use console.log calls within this component (and its children), since it will trigger an infinite loop!
+ */
+export default function Console() {
+  const classes = useStyles();
+
+  const [logs, setLogs] = useState([]);
+  const [searchValue, setSearchValue] = useState();
+  // See the constants for console methods defined in consoleHeader.js. If leaving empty, it will display all log entries
+  const [filters, setFilters] = useState(['error']);
+  const [direction, setDirection] = useState('descending');
+
+  useEffect(() => {
+    // eslint-disable-next-line arrow-parens
+    Hook(window.console, log => {
```

Comment:
instead of using a hook on console.log, I suggest using a prop or expose a `logger` class that is hooked to
https://github.com/tmrowco/tmrowapp-contrib/blob/master/playground/server/index.js#L77-L81
That will make the interface cleaner

Reply:
Yep makes sense! I will go the props way. Expanding the main app state was actually my first shot at it, but then decided to have only within the console component - Did not look careful enough at the `runLogs` listener!